x-pack/legacy/plugins/ml/server/routes/system.ts

```diff
@@ -11,20 +11,22 @@ import { RequestHandlerContext } from 'kibana/server';
 import { wrapError } from '../client/error_wrapper';
 import { mlLog } from '../client/log';
 import { privilegesProvider } from '../lib/check_privileges';
-import { isSecurityDisabled } from '../lib/security_utils';
 import { spacesUtilsProvider } from '../lib/spaces_utils';
-import { licensePreRoutingFactory } from '../new_platform/licence_check_pre_routing_factory';
+import { licensePreRoutingFactory } from '../new_platform/license_check_pre_routing_factory';
 import { RouteInitialization } from '../new_platform/plugin';
 
 /**
  * System routes
  */
 export function systemRoutes({
+  getLicenseCheckResults,
   router,
-  xpackMainPlugin,
+  xpackMainPlugin, // TODO: this needs to be replaced with NP
   spacesPlugin,
   cloud,
 }: RouteInitialization) {
+  const { isSecurityDisabled } = getLicenseCheckResults();
```

Comment:
this needs to move into each necessary route. when it is called here it gets the initial value, before the license observable has been updated.